pkg/cloud/amazon/vault/vault_resources.go

```diff
@@ -0,0 +1,342 @@
+package vault
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/aws/aws-sdk-go/service/iam"
+	goformation "github.com/awslabs/goformation/cloudformation"
+	"github.com/awslabs/goformation/cloudformation/resources"
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/jenkins-x/jx/pkg/util"
+)
+
+const stackName = "jenkins-cloud-formation-gitops-stack"
+
+// CreateVaultResources will automatically create the preresiquites for vault on aws
+func CreateVaultResources(region string, domain string, username string, tableName string) (*string, *string, *string, *string, error) {
```

Comment:
Why not returning just `string` instead of `*string`?

Reply:
So I cant return nil when I return `string`, that is why I resorted to `*string`